src/editors/containers/VideoUploadEditor/index.test.jsx

```diff
@@ -0,0 +1,113 @@
+  <IntlProvider locale="en">
+    <VideoUploadEditor {...props} />
+  </IntlProvider>,
+);
+
+const renderUploaderComponent = (props = defaultUploaderProps) => render(
+  <IntlProvider locale="en">
+    <VideoUploader {...props} />
+  </IntlProvider>,
+);
+
+describe('VideoUploadEditor', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders without errors', () => {
+    const { container } = renderEditorComponent();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders the input field for pasting the video ID or URL', () => {
+    const { getByPlaceholderText } = renderEditorComponent();
+    const input = getByPlaceholderText('Paste your video ID or URL');
+    expect(input).toBeInTheDocument();
+  });
+
+  it('updates the input field value when user types', () => {
+    const { getByPlaceholderText } = renderEditorComponent();
+    const input = getByPlaceholderText('Paste your video ID or URL');
+
+    fireEvent.change(input, { target: { value: 'test value' } });
+    expect(input.value).toBe('test value');
+  });
+
+  it('shows error message with unsupported files', async () => {
+    const { getByTestId, findByText } = renderEditorComponent();
+    const fileInput = getByTestId('fileInput');
+
+    const unsupportedFile = new File(['(⌐□_□)'], 'unsupported.avi', { type: 'video/avi' });
+    fireEvent.change(fileInput, { target: { files: [unsupportedFile] } });
+
+    const errorMsg = await findByText('Video must be an MP4 or MOV file');
+    expect(errorMsg).toBeInTheDocument();
+  });
+});
+
+describe('VideoUploader', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders without errors', () => {
+    const { container } = renderUploaderComponent();
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders with an error message', () => {
+    const errorMessage = 'Video must be an MP4 or MOV file';
+    const { getByText } = renderUploaderComponent({ ...defaultUploaderProps, errorMessage });
+    expect(getByText(errorMessage)).toBeInTheDocument();
+  });
+
+  it('handles unsupported video file formats', () => {
+    const { getByTestId } = renderUploaderComponent();
+    const fileInput = getByTestId('fileInput');
+    const unsupportedFile = new File(['(⌐□_□)'], 'unsupported.avi', { type: 'video/avi' });
+
+    fireEvent.change(fileInput, { target: { files: [unsupportedFile] } });
+    expect(mockOnUpload).not.toHaveBeenCalled();
+  });
+
+  it('calls the onUpload function when a supported file is dropped', async () => {
+    const { container } = renderUploaderComponent();
+    const dropzone = container.querySelector('.dropzone-middle');
+    const file = new File(['(⌐□_□)'], 'video.mp4', { type: 'video/mp4' });
+
+    await act(async () => {
+      fireEvent.drop(dropzone, {
+        dataTransfer: {
+          files: [file],
+          types: ['Files'],
+        },
+      });
+    });
+
+    expect(mockOnUpload).toHaveBeenCalledWith(file);
+  });
```

Comment:
@pkulkark  Can we also check if the selected file displayed in the UI?

Reply:
I removed displaying part because it was not in the designs. I think they'll be handled in a separate work flow.